### DISCUSSION TITLE
Set default code in frontend for alarm_control_panel

### DISCUSSION THREAD

**Discussion by gjohansson-ST** [home-assistant/architecture#892] [Entity Models] [answered]
### Story

Similar to the implementation of a default code for [locks](https://github.com/home-assistant/architecture/discussions/870) we should do same for alarm_control_panel.

### Proposal

- Implement a user customization for `alarm_control_panel` so that a user can set a default code for a certain `alarm_control_panel` entity in frontend.
- Implement validation similar to `lock` entity but also consider `attr_code_arm_required`.

### Usage

- Service calls are updated so if default code is set, code in service calls are optional as it will use default code if no code is supplied (if validation requires a code)
- Frontend is updated (`alarm_control_panel` card) so if default code is set, it's possible to click arm/disarm without supplying a code if required as it will then use the default code.

### Other

- Integrations currently setting a default code as part of config entry options can now remove this.

### Result

Users can now directly in frontend for the specific `alarm_control_panel` entity provide a default `code` which will be considered in validation and thereby `code` can be omited in service calls /frontend even if `code_format` has been set to validate the `code`.

### Reference PR's
To be created

**Comment by emontnemery** [comment, marked as answer]
This proposal was discussed and approved by the core team, please go ahead with the implementation @gjohansson-ST 👍

**Reply by gjohansson-ST** [reply]
Implemented in https://github.com/home-assistant/core/pull/112540

**Comment by SpectrumTech** [comment]
And it broke my alarm panel interpretation and I have no work around as my integration has no unique ID… Why are you forcing people to use an alarm code? Who breaks in your house and sets the alarm? Who breaks in, and figures out your HA integration quick enough to disarm your alarm? Most crooks will flee at the sound of an alarm… What technology advanced dystopia do you live in? The default was fine. Either make your case the edge case or implement a workaround so everyone else can proceed as usual. At least add the “code_required” Boolean to configuration so I can go back to living my life!

**Reply by frenck** [reply]
You might want to reconsider your tone just a little bit... Everyone tries to improve things here, and no one has the intention to break things. Above all, we are all humans here too.

> And it broke my alarm panel interpretation

If this broke an integration, please raise an issue in the issue tracker that provides this integration. In case it is a built-in integration, shipped by Home Assistant by default, please report it here:

<https://github.com/home-assistant/core/issues>

Otherwise, report it to the custom integration author's repository.

../Frenck

**Comment by DaveCBr** [comment]
I get the change but do not like it.  I would much rather have the default code in Secrets than being able to easily expose the code in the GUI.  All it would take  is a person with very little HA knowledge to find the code.  I think that you should make this selectable.  Add a checkbox to select where the code is saved.

**Reply by frenck** [reply]
> I would much rather have the default code in Secrets than be able to easily expose the code in the GUI.

There is no difference between the two. The secret files exist to exclude them from a Git repository when publishing your automations to, for example, a public GitHub repo. The location used in this proposal isn't suitable for publishing either (storage folder) thus the result remains the same.

../Frenck
